requests_oauthlib/oauth2_session.py

```diff
@@ -197,10 +197,18 @@ def fetch_token(self, token_url, code=None, authorization_response=None,
                 redirect_uri=self.redirect_uri, username=username,
                 password=password, **kwargs)
 
-        if (not auth) and username:
-            if password is None:
-                raise ValueError('Username was supplied, but not password.')
-            auth = requests.auth.HTTPBasicAuth(username, password)
+        client_id = kwargs.get('client_id', '')
+        if not auth:
+            if client_id:
+                log.debug('Encoding client_id "%s" with client_secret as Basic auth credentials.', client_id)
+                client_secret = kwargs.get('client_secret', '')
+                client_secret = client_secret if client_secret is not None else ''
```

Comment:
Mmm, yeah, so I was running under the assumption that `client_secret` wouldn't be passed as `None`, but now that you raise the possibility it occurs to me that I think some calls may default it that way. Let's leave this code in as defense-in-depth.